File: 프로그래머스/완전탐색/카펫.py
def solution(brown, yellow):
    answer = []
    
    n = brown + yellow
    
    for i in range(3, int(n/2)+1):
        j = n / i
        
        if (j*10) % 10 != 0 : continue
        
        if (i-2) * (j-2) == yellow:
            answer.append(i)
            answer.append(int(j))
            break
            
    if answer[0] < answer[1]:
        answer.sort(reverse=True)
    
    return answer

    # result = brown + yellow
    # n = result // 2
    
    # data = []
    # for i in range(1, n+1):
    #     if result % i == 0: 
    #         data.append(i)
    #         data.append(result//i) 
    # data = sorted(list(set(data)))
    
    # l = len(data)
    # if l%2 == 0: answer = [data[l//2], data[l//2-1]]
    # else: answer = [data[l//2], data[l//2]]
    
    # return answer

File: 프로그래머스/문제_lv2/카펫.js
function solution(brown, yellow) {
  var answer = [];

  // 2X^2 + (4-brown)X + 2yellow = 0
  const x =
    ((brown - 4) / 2 + Math.sqrt(Math.pow((brown - 4) / 2, 2) - 4 * yellow)) /
    2;
  const y = yellow / x;

  answer = [Math.max(x + 2, y + 2), Math.min(x + 2, y + 2)];

  return answer;
}
